### ISSUE TITLE
72e6bbfa9f0a4 breaks startup

### ISSUE THREAD

**Issue by chdiza** [macvim-dev/macvim#928]
I just built 72e6bbfa9f0a4 and ran it.

Upon opening a new MacVim window, now there is no "splash screen" with the version on it.  I.e., the part that starts with:
```

~                              VIM - Vi IMproved
~
~                               version 8.1.1770
~                           by Bram Moolenaar et al.
~                 Vim is open source and freely distributable
```

**Comment by chdiza** [comment]
The splash screen is not suppressed if I run the `Vim` binary that's inside MacVim.app in the console.

**Comment by chdiza** [comment]
It really is that most recent merge that did it; fc11e4bf80c3 is completely fine.

**Comment by ychin** [comment]
@chdiza what OS are you using? Also do you have custom fonts? It is working fine for me in 10.14 with default fonts.

**Comment by chdiza** [comment]
I'm on 10.14.6, and my .gvimrc sets [Input Mono Light](https://input.fontbureau.com) as `&gfn`.  This is not some garbage-ified "powerline" version of the font either.  When you download that font they let you select some parameters like whether the zero is slashed, but I didn't touch anything having to do with descenders or heights.

Sometimes I can see the "splash" screen happen *very* quickly and then disappear, even though I pressed no keys.

Also, once I start editing a document the interline spacing is all messed up, namely it's gigantic (what I would call almost double-spacing if this were a WYSIWYG word processor).  This font does *not* produce that result in iTerm2.app or Terminal.app or BBEdit.app.

I've tested a few other fonts and they seem to be OK, except that a few that I rarely use also seem to have unexpectedly tall interline spaces (e.g., Monaco 12pt, though I might just be imagining the tall spacing with this one).

**Comment by chdiza** [comment]
I tried it on 10.13.6 today just to see if that's relevant, and there's a small difference.  The splashscreen does show up, but the line spacing is still comically large.

**Comment by ychin** [comment]
I think the issue is that the font you are using has a >1 line spacing setting that the previous code wasn't using. The new streamlined code just takes the NSFont in and use it which is probably why it respects the line spacing setting now (I would imagine this is also why the old code has a convoluted way to create a temporary font).

I checked other text editors and seems like it's a mix. Some editors like Xcode and Sublime Text seems to respect the line spacing, while Visual Studio Code ignores. Since this changes MacVim's behavior I should probably revert it, but I'll take a 2nd pass at it first to think about it a little.

**Comment by chdiza** [comment]
OK thanks for looking into it.  I would think the important thing regarding line spacing is to have MacVim match whatever Vim does, and Vim is used in terminals, and both iTerm and Terminal.app don't have the gigantic line spacing.  (How weird it is to like a particular coding font but have it be unusable in MacVim yet fine in Vim!)

The problem of the vanished splash screen, on the other hand, is surely a bug and all by itself warrants a revert.

**Comment by ychin** [comment]
> The problem of the vanished splash screen, on the other hand, is surely a bug and all by itself warrants a revert.

This is just a symptom of the new linespace causing MacVim to attempt to create too large a window (since it remembers your last stored number of columns and rows, see `defaults read org.vim.MacVim MMAutosaveRows` and `defaults read org.vim.MacVim MMAutosaveColumns`) and had to resize down right at launch, which makes the splash screen disappear.

It's not a bug per se.

**Comment by ychin** [comment]
> OK thanks for looking into it. I would think the important thing regarding line spacing is to have MacVim match whatever Vim does, and Vim is used in terminals, and both iTerm and Terminal.app don't have the gigantic line spacing. (How weird it is to like a particular coding font but have it be unusable in MacVim yet fine in Vim!)

Actually, I think iTerm or Terminal *do* respect line spacing of the font, producing wide gaps. Not sure why you didn't see that but for my local testing with a large line spacing fonts those two did seem to make pretty tall lines.

**Comment by chdiza** [comment]
> Actually, I think iTerm

Nope, not with `Input Mono`.  Maybe they do with *other* tall fonts.

**Comment by ychin** [comment]
Actually, @chdiza can you share screenshots and your guifont settings in Vimrc? For Input font do you just download the stock font with no modifications? (You can customize the font by modifying line space etc)

I just realized I was testing incorrectly, and now I can't seem to see the issue you are quoting, and it's hard to tell without seeing a screenshot or knowing exactly what font and size you are setting to.

**Comment by ychin** [comment]
The issue should be fixed now, I think. Let's me know if you see any issues @chdiza since I couldn't quite repro it myself.

**Comment by chdiza** [comment]
@ychin It does seem to be fixed.  Thanks!
